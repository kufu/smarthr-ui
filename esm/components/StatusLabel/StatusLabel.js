var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import React from 'react';
import styled, { css } from 'styled-components';
import { useTheme } from '../../hooks/useTheme';
export var StatusLabel = function (_a) {
    var _b = _a.type, type = _b === void 0 ? 'done' : _b, _c = _a.className, className = _c === void 0 ? '' : _c, children = _a.children;
    var theme = useTheme();
    return (React.createElement(Wrapper, { className: type + " " + className, themes: theme }, children));
};
var borderStyle = function (border) { return css(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  border: ", ";\n  background-color: #fff;\n"], ["\n  border: ", ";\n  background-color: #fff;\n"])), border); };
var fillStyle = function (backgroundColor) { return css(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n  background-color: ", ";\n  color: #fff;\n"], ["\n  background-color: ", ";\n  color: #fff;\n"])), backgroundColor); };
var Wrapper = styled.span(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n  ", "\n"], ["\n  ",
    "\n"])), function (_a) {
    var themes = _a.themes;
    var size = themes.size, frame = themes.frame, palette = themes.palette;
    return css(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n      height: ", ";\n      box-sizing: border-box;\n      margin: 0;\n      padding: 0 ", ";\n      display: inline-block;\n      white-space: nowrap;\n      font-size: ", ";\n      font-weight: bold;\n      line-height: calc(", " - ", " * 2);\n\n      &.done {\n        ", "\n        border-color: ", ";\n        color: ", ";\n      }\n\n      &.success {\n        ", "\n        border-color: ", ";\n        color: ", ";\n      }\n\n      &.process {\n        ", "\n        border-color: ", ";\n        color: ", ";\n      }\n\n      &.required {\n        ", "\n        border-color: ", ";\n        color: ", ";\n      }\n\n      &.disabled {\n        ", "\n        background-color: ", ";\n      }\n\n      &.must {\n        ", "\n        background-color: ", ";\n      }\n\n      &.warning {\n        ", "\n        background-color: ", ";\n      }\n\n      &.error {\n        ", "\n        background-color: ", ";\n      }\n    "], ["\n      height: ", ";\n      box-sizing: border-box;\n      margin: 0;\n      padding: 0 ", ";\n      display: inline-block;\n      white-space: nowrap;\n      font-size: ", ";\n      font-weight: bold;\n      line-height: calc(", " - ", " * 2);\n\n      &.done {\n        ", "\n        border-color: ", ";\n        color: ", ";\n      }\n\n      &.success {\n        ", "\n        border-color: ", ";\n        color: ", ";\n      }\n\n      &.process {\n        ", "\n        border-color: ", ";\n        color: ", ";\n      }\n\n      &.required {\n        ", "\n        border-color: ", ";\n        color: ", ";\n      }\n\n      &.disabled {\n        ", "\n        background-color: ", ";\n      }\n\n      &.must {\n        ", "\n        background-color: ", ";\n      }\n\n      &.warning {\n        ", "\n        background-color: ", ";\n      }\n\n      &.error {\n        ", "\n        background-color: ", ";\n      }\n    "])), size.pxToRem(20), size.pxToRem(size.space.XXS), size.pxToRem(size.font.SHORT), size.pxToRem(20), frame.border.lineWidth, borderStyle(frame.border.default), palette.BORDER, palette.TEXT_GREY, borderStyle(frame.border.default), palette.MAIN, palette.MAIN, borderStyle(frame.border.default), palette.WARNING, palette.WARNING, borderStyle(frame.border.default), palette.DANGER, palette.DANGER, fillStyle(palette.BORDER), palette.TEXT_GREY, fillStyle(palette.BORDER), palette.MAIN, fillStyle(palette.BORDER), palette.WARNING, fillStyle(palette.BORDER), palette.DANGER);
});
var templateObject_1, templateObject_2, templateObject_3, templateObject_4;
//# sourceMappingURL=StatusLabel.js.map