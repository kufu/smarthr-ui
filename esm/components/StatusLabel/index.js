export { StatusLabel } from './StatusLabel';
//# sourceMappingURL=index.js.map