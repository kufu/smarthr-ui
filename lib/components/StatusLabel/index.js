"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StatusLabel = void 0;
var StatusLabel_1 = require("./StatusLabel");
Object.defineProperty(exports, "StatusLabel", { enumerable: true, get: function () { return StatusLabel_1.StatusLabel; } });
//# sourceMappingURL=index.js.map