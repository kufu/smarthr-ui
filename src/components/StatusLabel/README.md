# StatusLabel

```tsx
import { StatusLabel } from 'smarthr-ui'

<StatusLabel type="require">必須</StatusLabel>
```

## props

| Name       | Required | Type                              | DefaultValue | Description                                                       |
| ---------- | -------- | --------------------------------- | ------------ | ----------------------------------------------------------------- |
| skeleton     | -        | **boolean** <br> true &#124; false | false            | Set transparent background when its true  |
| type | ✓           | **enum** <br> `'done'` &#124; `'success'` &#124; `'process'` &#124; `'required'` &#124; `'disabled'` &#124; `'must'` &#124; `'warning'` &#124; `'error'` | 'done'            | Can be set type of tag   |
