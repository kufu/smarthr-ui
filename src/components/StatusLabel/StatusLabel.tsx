import React, { FC } from 'react'
import styled, { css } from 'styled-components'

import { Theme, useTheme } from '../../hooks/useTheme'

type Props = {
  type?: 'done' | 'success' | 'process' | 'required' | 'disabled' | 'must' | 'warning' | 'error'
  className?: string
  children: string
}

export const StatusLabel: FC<Props> = ({ type = 'done', className = '', children }) => {
  const theme = useTheme()

  return (
    <Wrapper className={`${type} ${className}`} themes={theme}>
      {children}
    </Wrapper>
  )
}

const borderStyle = (border: string) => css`
  border: ${border};
  background-color: #fff;
`
const fillStyle = (backgroundColor: string) => css`
  background-color: ${backgroundColor};
  color: #fff;
`
const Wrapper = styled.span<{ themes: Theme }>`
  ${({ themes }) => {
    const { size, frame, palette } = themes

    return css`
      height: ${size.pxToRem(20)};
      box-sizing: border-box;
      margin: 0;
      padding: 0 ${size.pxToRem(size.space.XXS)};
      display: inline-block;
      white-space: nowrap;
      font-size: ${size.pxToRem(size.font.SHORT)};
      font-weight: bold;
      line-height: calc(${size.pxToRem(20)} - ${frame.border.lineWidth} * 2);

      &.done {
        ${borderStyle(frame.border.default)}
        border-color: ${palette.BORDER};
        color: ${palette.TEXT_GREY};
      }

      &.success {
        ${borderStyle(frame.border.default)}
        border-color: ${palette.MAIN};
        color: ${palette.MAIN};
      }

      &.process {
        ${borderStyle(frame.border.default)}
        border-color: ${palette.WARNING};
        color: ${palette.WARNING};
      }

      &.required {
        ${borderStyle(frame.border.default)}
        border-color: ${palette.DANGER};
        color: ${palette.DANGER};
      }

      &.disabled {
        ${fillStyle(palette.BORDER)}
        background-color: ${palette.TEXT_GREY};
      }

      &.must {
        ${fillStyle(palette.BORDER)}
        background-color: ${palette.MAIN};
      }

      &.warning {
        ${fillStyle(palette.BORDER)}
        background-color: ${palette.WARNING};
      }

      &.error {
        ${fillStyle(palette.BORDER)}
        background-color: ${palette.DANGER};
      }
    `
  }}
`
