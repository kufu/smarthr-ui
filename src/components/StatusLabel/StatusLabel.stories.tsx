import { storiesOf } from '@storybook/react'
import * as React from 'react'
import styled from 'styled-components'

import { StatusLabel } from './StatusLabel'
import readme from './README.md'

storiesOf('StatusLabel', module)
  .addParameters({
    readme: {
      sidebar: readme,
    },
  })
  .add('all', () => (
    <ListWrapper>
      <List>
        <li>
          <StatusLabel type="done">完了</StatusLabel>
        </li>
        <li>
          <StatusLabel type="success">入力完了</StatusLabel>
        </li>
        <li>
          <StatusLabel type="process">入力中</StatusLabel>
        </li>
        <li>
          <StatusLabel type="required">入力必要</StatusLabel>
        </li>
      </List>
      <List>
        <li>
          <StatusLabel type="disabled">不可</StatusLabel>
        </li>
        <li>
          <StatusLabel type="must">必読</StatusLabel>
        </li>
        <li>
          <StatusLabel type="warning">警告</StatusLabel>
        </li>
        <li>
          <StatusLabel type="error">エラー</StatusLabel>
        </li>
      </List>
    </ListWrapper>
  ))

const ListWrapper = styled.div`
  padding: 0 24px;
`

const List = styled.ul`
  display: flex;
  padding: 0;
  margin: 0 0 24px;

  > li {
    display: block;
    margin-right: 16px;
  }
`
