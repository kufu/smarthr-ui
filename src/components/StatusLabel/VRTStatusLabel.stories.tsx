import { StoryFn } from '@storybook/react'
import React from 'react'
import styled from 'styled-components'

import { InformationPanel } from '../InformationPanel'

import { StatusLabel } from './StatusLabel'
import { All } from './StatusLabel.stories'

export default {
  title: 'States（状態）/StatusLabel',
  component: StatusLabel,
  parameters: {
    withTheming: true,
  },
}

export const VRTStatusLabelForcedColors: StoryFn = () => (
  <Wrapper>
    <VRTInformationPanel title="VRT 用の Story です" togglable={false}>
      Chromatic 上では強制カラーモードで表示されます
    </VRTInformationPanel>
    <All />
  </Wrapper>
)

VRTStatusLabelForcedColors.parameters = {
  chromatic: { forcedColors: 'active' },
}

const Wrapper = styled.div`
  height: 100vh;
  box-sizing: border-box;
  padding: 24px;
  color: ${({ theme }) => theme.color.TEXT_BLACK};
`

const VRTInformationPanel = styled(InformationPanel)`
  margin-bottom: 24px;
`
