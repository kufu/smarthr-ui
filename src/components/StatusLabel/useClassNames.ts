import { useMemo } from 'react'

import { useClassNameGenerator } from '../../hooks/useClassNameGenerator'
import { StatusLabel } from './StatusLabel'

export function useClassNames() {
  const generate = useClassNameGenerator(StatusLabel.displayName || 'StatusLabel')
  return useMemo(
    () => ({
      wrapper: generate(),
    }),
    [generate],
  )
}
