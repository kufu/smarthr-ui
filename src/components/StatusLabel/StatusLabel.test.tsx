import { create } from 'react-test-renderer'
import React from 'react'
import ReactDOM from 'react-dom'

import { StatusLabel } from './StatusLabel'

describe('StatusLabel', () => {
  let container: HTMLDivElement
  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })
  afterEach(() => {
    document.body.removeChild(container)
  })
  const hello = 'hello'
  it('should be match snapshot', () => {
    const testRenderer = create(<StatusLabel type="success">{hello}</StatusLabel>)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
  it('should render given children', () => {
    ReactDOM.render(<StatusLabel type="success">{hello}</StatusLabel>, container)
    expect(container.textContent).toBe(hello)
  })
  it('should have given type', () => {
    const testInstance = create(<StatusLabel type="success">{hello}</StatusLabel>).root
    expect(testInstance.props.type).toBe('success')
  })
})
