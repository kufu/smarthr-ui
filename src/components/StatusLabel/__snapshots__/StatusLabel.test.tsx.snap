// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusLabel should be match snapshot 1`] = `
.c0 {
  height: 1.25rem;
  box-sizing: border-box;
  margin: 0;
  padding: 0 0.5rem;
  display: inline-block;
  white-space: nowrap;
  font-size: 0.6875rem;
  font-weight: bold;
  line-height: calc(1.25rem - 1px * 2);
}

.c0.done {
  border: 1px solid #d6d6d6;
  background-color: #fff;
  border-color: #d6d6d6;
  color: #767676;
}

.c0.success {
  border: 1px solid #d6d6d6;
  background-color: #fff;
  border-color: #00a5ab;
  color: #00a5ab;
}

.c0.process {
  border: 1px solid #d6d6d6;
  background-color: #fff;
  border-color: #ff8800;
  color: #ff8800;
}

.c0.required {
  border: 1px solid #d6d6d6;
  background-color: #fff;
  border-color: #ef475b;
  color: #ef475b;
}

.c0.disabled {
  background-color: #d6d6d6;
  color: #fff;
  background-color: #767676;
}

.c0.must {
  background-color: #d6d6d6;
  color: #fff;
  background-color: #00a5ab;
}

.c0.warning {
  background-color: #d6d6d6;
  color: #fff;
  background-color: #ff8800;
}

.c0.error {
  background-color: #d6d6d6;
  color: #fff;
  background-color: #ef475b;
}

<span
  className="c0 success "
>
  hello
</span>
`;
